test/Control/Applicative/CompatSpec.hs

```diff
@@ -30,6 +30,24 @@ spec = do
       getConst ((Const "aaa" :: Const String Int) <> (Const "bbb" :: Const String Int))
         `shouldBe` "aaabbb"
 
+  describe "Eq (Const a b)" $
+    it "allows checking if two Const values are equal" $ do
+      (Const "a" :: Const String Int) == Const "a" `shouldBe` True
+      (Const "a" :: Const String Int) == Const "b" `shouldBe` False
+
+  describe "Ord (Const a b)" $
+    it "allows comparing of two Const values" $ do
+      (Const 'b' :: Const Char Int) >  Const 'a' `shouldBe` True
+      (Const 'b' :: Const Char Int) <= Const 'a' `shouldBe` False
+
+  describe "Read (Const a b)" $
+    it "allows a Const value to be read from a string" $
+      read "Const 1" `shouldBe` (Const 1 :: Const Int Int)
+
+  describe "Show (Const a b)" $
+    it "allows a Const value to be converted to a String" $
+      show (Const 1 :: Const Int Int) `shouldBe` "Const 1"
+
```

Comment:
I'm on the fence for writing specs for derived code.  It's nice to ensure that instances behave the same across multiple versions of GHC, but on the other hand we actually test that GHC's deriving mechanism works here.

If I would write it, I'm still undecided whether I want to have specs in cases like that or not.